host/xtest/asym_perf.c

```diff
@@ -0,0 +1,962 @@
+	case RSASSA_PKCS1_PSS_MGF1_SHA1:
+		return SHA1_LEN;
+	case RSAES_PKCS1_OAEP_SHA224:
+	case RSASSA_PKCS1_V1_5_SHA224:
+	case RSASSA_PKCS1_PSS_MGF1_SHA224:
+		return SHA224_LEN;
+	case RSAES_PKCS1_OAEP_SHA256:
+	case RSASSA_PKCS1_V1_5_SHA256:
+	case RSASSA_PKCS1_PSS_MGF1_SHA256:
+		return SHA256_LEN;
+	case RSAES_PKCS1_OAEP_SHA384:
+	case RSASSA_PKCS1_V1_5_SHA384:
+	case RSASSA_PKCS1_PSS_MGF1_SHA384:
+		return SHA384_LEN;
+	case RSAES_PKCS1_OAEP_SHA512:
+	case RSASSA_PKCS1_V1_5_SHA512:
+	case RSASSA_PKCS1_PSS_MGF1_SHA512:
+		return SHA512_LEN;
+	default:
+		fprintf(stderr, "The algo[%u] is not valid!\n", algo);
+	}
+
+	return -1;
+}
+
+static int check_rsa_cipher_params(uint32_t crypto_algo, int width_bits, int size)
+{
+	int width_bytes = BITS_TO_BYTES(width_bits);
+	int hash_len = 0;
+
+	if (crypto_algo == RSA_NOPAD) {
+		if (size > width_bytes) {
+			fprintf(stderr, "The size or algo is not valid!\n");
+			return -1;
+		}
+	} else if (crypto_algo == RSAES_PKCS1_V1_5) {
+		if ((size + PKCS_V1_5_MIN) > width_bytes) {
+			fprintf(stderr, "The size or algo is not valid!\n");
+			return -1;
+		}
+	} else if (crypto_algo > RSAES_PKCS1_V1_5) {
+		hash_len = get_rsa_hash_len(crypto_algo);
+		if (hash_len == -1)
+			return -1;
+
+		if (OAEP_HASH_LEN(hash_len) >= (width_bytes - OAEP_OTHER_LEN)) {
+			fprintf(stderr, "The width_bits or algo is not valid!\n");
+			return -1;
+		} else if (size > (width_bytes - OAEP_HASH_LEN(hash_len) -
+			   OAEP_OTHER_LEN)) {
+			fprintf(stderr, "The size or algo is not valid!\n");
+			return -1;
+		}
+	} else {
+		return -1;
+	}
+
+	return 0;
+}
+
+static int check_rsa_hash_params(uint32_t crypto_algo, int width_bits, int size,
+				 int salt_len)
+{
+	int width_bytes = BITS_TO_BYTES(width_bits);
+	int salt_temp = 0;
+	int hash_len = get_rsa_hash_len(crypto_algo);
+
+	if (hash_len == -1)
+		return -1;
+
+	switch (crypto_algo) {
+	case RSASSA_PKCS1_V1_5_SHA1:
+		if (width_bytes < hash_len + DERCODE_SHA1_LEN + PKCS_V1_5_MIN) {
+			fprintf(stderr, "The size or algo is not valid!\n");
+			return -1;
+		}
+		return 0;
+	case RSASSA_PKCS1_V1_5_SHA224:
+	case RSASSA_PKCS1_V1_5_SHA256:
+	case RSASSA_PKCS1_V1_5_SHA384:
+	case RSASSA_PKCS1_V1_5_SHA512:
+		if (width_bytes < hash_len + DERCODE_SHA_LEN + PKCS_V1_5_MIN) {
+			fprintf(stderr, "The size or algo is not valid!\n");
+			return -1;
+		}
+		return 0;
+	case RSASSA_PKCS1_PSS_MGF1_SHA1:
+	case RSASSA_PKCS1_PSS_MGF1_SHA224:
+	case RSASSA_PKCS1_PSS_MGF1_SHA256:
+	case RSASSA_PKCS1_PSS_MGF1_SHA384:
+	case RSASSA_PKCS1_PSS_MGF1_SHA512:
+		salt_temp = (salt_len == 0 ? hash_len : salt_len);
+		if (salt_temp > width_bytes ||
+		    width_bytes < hash_len + salt_temp + PSS_OTHER_LEN) {
+			fprintf(stderr, "The size or algo is not valid!\n");
+			return -1;
+		}
+		return 0;
+	default:
+		return -1;
+	}
+}
+
+int asym_perf_runner_cmd_parser(int argc, char *argv[])
+{
+	int verbosity = CRYPTO_DEF_VERBOSITY;
+	int is_random = CRYPTO_USE_ZEROS;
+	unsigned int n = CRYPTO_DEF_COUNT;
```

Comment:
These `n` and `l` values are remarkably high for the selected default algo.
I would suggest to lower these values.